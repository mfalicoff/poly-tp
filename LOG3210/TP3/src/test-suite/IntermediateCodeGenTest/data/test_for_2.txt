Declaration
num a;
num i;
num j;
EndDeclaration

a = 0;

for (i = 0; i < 3; i = i + 1) {
  for (j = 0; j < 3; j = j + 2) {
    a = i;
  }
}
