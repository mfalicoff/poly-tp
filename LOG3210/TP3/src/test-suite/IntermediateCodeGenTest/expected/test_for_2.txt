a = 0
_L1
i = 0
_L2
if i < 3 goto _L4
goto _L0
_L4
j = 0
_L5
if j < 3 goto _L7
goto _L3
_L7
a = i
_L6
_t0 = j + 2
j = _t0
goto _L5
_L3
_t1 = i + 1
i = _t1
goto _L2
_L0
